src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -448,6 +448,14 @@ private BoundStatement BindLocalFunctionStatement(LocalFunctionStatementSyntax n
             var hasErrors = localSymbol.ScopeBinder
                 .ValidateDeclarationNameConflictsInScope(localSymbol, diagnostics);
 
+            foreach (var p in node.ParameterList.Parameters)
+            {
+                if (p.IsArgList)
+                {
+                    diagnostics.Add(ErrorCode.ERR_IllegalVarArgs, p.Location);
```

Comment:
>diagnostics.Add(ErrorCode.ERR_IllegalVarArgs, p.Location); [](start = 20, length = 58)

It feels like this check belongs in GrabDiagnostics/GetDeclarationDiagnostics on the local function symbol.